### PR TITLE
Zoom page content

### PR DESCRIPTION
**Pull request by intermittentnrg**
I originally sent ctrl+ to browser to zoom, but the screenshot unzooms so I found this method instead.

window size and zoom factor should be provided as options obviously...

### CODE REVIEW

**Review by amotl — COMMENTED** [review]
Excellent. I've not run the program yet, to get an idea about the outcome, but I like your attention to details.

Shall I support you on bringing in the zoom factor as option/setting?

**Comment by intermittentnrg** [comment]
Yes support would be fantastic thank you!

I've had a lot of feedback that people couldn't read the text when posting widescreen videos on X/Twitter, and given the styling/studio settings do not work this is my solution which works quite well. I'm not even sure of the expected result from the styling as I've not seen it work...

2) An outstanding issue from feedback is that the labels on the X axis jump around. Just changing dateformat probably improves it, it annoyingly can only be set globally via grafana.ini and it messes up my normal use of grafana.

3) I also hacked timeutils.py to suit my needs which also needs to be made an option.
```
def get_freq_delta(every: str) -> RecurrenceInfo:
    rr_interval = 4
    rr_freq = HOURLY
    delta = timedelta(days=30) # - timedelta(seconds=1)
    return RecurrenceInfo(every=every, frequency=rr_freq, interval=rr_interval, duration=delta)
```
Not sure your usecase was for timeseries panels? Or I may have misunderstood the options. But this was ideal for me to set step size and time range shown.

**Comment by amotl** [comment]
Hi @maurerle,

it looks like @intermittentnrg also contributed more valuable improvements here, both within the PR, and possibly also about the `get_freq_delta` function? If you are still in the right mood, would you be keen on wrapping this up?

With kind regards,
Andreas.

**Comment by maurerle** [comment]
Hi @amotl
Thanks for taking care of the release!
I am currently quite busy with other things, but if I have some more free time again and needs to create new graphics, I will pursue this further :)

But I am more than happy that we got something working with latest grafana for now :)
I did not have the need to zoom panels with grafana yet, so I am still unsure if we should add yet another config param for this

File: grafanimate/grafana.py [hunk 1]
```diff
@@ -38,12 +42,17 @@ def boot_grafana(self):
         Navigate to Grafana application and inject Grafana Sidecar service.
         """
         log.info("Starting Grafana at %s", self.baseurl)
-        self.set_window_size(1920, 1080)
-
+        if self.window_size:
+            self.set_window_size(
+                int(self.window_size[0]),
+                int(self.window_size[1] + (85 * self.zoom_factor)),
+            )
```
Comment (amotl) [inline grafanimate/grafana.py:49]: Is that ` height + 85 * self.zoom_factor` formular correct? Can you elaborate, why "85"?

**Review by amotl — COMMENTED** [review]
Hi @intermittentnrg,

I've extended your patch to make relevant new options configurable using `--window-size` and `--zoom-factor`, and added one question about its ingredients.

With kind regards,
Andreas.

**Comment by intermittentnrg** [comment]
85 was the height of the firefox chrome (location bar etc) and was being zoomed together with the page using this method if I recall correctly.

**Comment by amotl** [comment]
I see. Thanks for clarifying.
So, let's merge and include the improvement into the next release, if you don't have any objections?

**Review by maurerle — APPROVED** [review]

**Pull request merged**